templates/explore/repo_list.tmpl

```diff
@@ -54,9 +54,9 @@
 				{{$description := .DescriptionHTML $.Context}}
 				{{if $description}}<p>{{$description}}</p>{{end}}
 				{{if .Topics}}
-					<div class="ui tags">
+					<div class="ui tags" id="repo-topics">
```

Comment:
It's not right. You are generating duplicate "ID" in the `range` loop.

Reply:
Can I remove id="repo-topics" and add class="repo-topics"?

Reply:
I haven't tried, but I think you can use flex layout. eg `gt-df gt-gap-2`

Reply:
I want to try the following code
```html
{{if .Topics}}
	<div class="ui tags repo-topics">
	{{range .Topics}}
		{{if ne . ""}}<a href="{{AppSubUrl}}/explore/repos?q={{.}}&topic=1"><div class="ui repo-topic small label topic">{{.}}</div></a>{{end}}
	{{end}}
	</div>
{{end}}
```
and
```css
.repo-topics .repo-topic {
  font-weight: var(--font-weight-normal);
  cursor: pointer;
  margin: 2px;
}
```

Reply:
The simplest change (from main branch) is this:

```diff
diff --git a/templates/explore/repo_list.tmpl b/templates/explore/repo_list.tmpl
index 49adfe5b4..e062c8bf9 100644
--- a/templates/explore/repo_list.tmpl
+++ b/templates/explore/repo_list.tmpl
@@ -56,7 +56,7 @@
                                {{if .Topics}}
                                        <div class="ui tags">
                                        {{range .Topics}}
-                                               {{if ne . ""}}<a href="{{AppSubUrl}}/explore/repos?q={{.}}&topic=1"><div class="ui small label topic">{{.}}</div></a>{{end}}
+                                               {{if ne . ""}}<a class="ui small label" href="{{AppSubUrl}}/explore/repos?q={{.}}&topic=1">{{.}}</a>{{end}}
                                        {{end}}
                                        </div>
                                {{end}}
diff --git a/web_src/css/explore.css b/web_src/css/explore.css
index f4d1eed96..8b1ef3ef8 100644
--- a/web_src/css/explore.css
+++ b/web_src/css/explore.css
@@ -33,6 +33,9 @@
 }
 
 .ui.repository.list .item .ui.tags {
+  display: flex;
+  gap: 0.25em;
+  flex-wrap: wrap;
   margin-bottom: 0.5rem;
 }
```

OR this (they are the same)

```diff
diff --git a/templates/explore/repo_list.tmpl b/templates/explore/repo_list.tmpl
index 49adfe5b4..8e872af19 100644
--- a/templates/explore/repo_list.tmpl
+++ b/templates/explore/repo_list.tmpl
@@ -54,9 +54,9 @@
                                {{$description := .DescriptionHTML $.Context}}
                                {{if $description}}<p>{{$description}}</p>{{end}}
                                {{if .Topics}}
-                                       <div class="ui tags">
+                                       <div class="gt-df gt-fw gt-gap-2 gt-mb-3">
                                        {{range .Topics}}
-                                               {{if ne . ""}}<a href="{{AppSubUrl}}/explore/repos?q={{.}}&topic=1"><div class="ui small label topic">{{.}}</div></a>{{end}}
+                                               {{if ne . ""}}<a class="ui small label" href="{{AppSubUrl}}/explore/repos?q={{.}}&topic=1">{{.}}</a>{{end}}
                                        {{end}}
                                        </div>
                                {{end}}
diff --git a/web_src/css/explore.css b/web_src/css/explore.css
index f4d1eed96..139dfcb19 100644
--- a/web_src/css/explore.css
+++ b/web_src/css/explore.css
@@ -32,10 +32,6 @@
   font-size: 12px;
 }
 
-.ui.repository.list .item .ui.tags {
-  margin-bottom: 0.5rem;
-}
-
 .ui.repository.list .repo-title .labels {
   word-break: normal;
   flex-shrink: 0;
```

![image](https://github.com/go-gitea/gitea/assets/2114189/14b5bffa-5c2d-4b70-a4ee-a67ef1347904)

Reply:
Yes, flexbox and `gap` is the right solution here, via helper classes `gt-gap-{1-5}`.